pkg/services/alerting/rule.go

```diff
@@ -113,9 +115,21 @@ func NewRuleFromDBAlert(ruleDef *m.Alert) (*Rule, error) {
 		jsonModel := simplejson.NewFromAny(v)
 		id, err := jsonModel.Get("id").Int64()
 		if err != nil {
-			return nil, ValidationError{Reason: "Invalid notification schema", DashboardId: model.DashboardId, Alertid: model.Id, PanelId: model.PanelId}
+			notificationName, notificationNameErr := jsonModel.Get("name").String()
```

Comment:
I was wrong about this. We need to support both `id` and `uid` in this case. Otherwise alerts created before this will not work.

But If the uid is missing we can assume id is in the old format the id as an uid using the fmt package. Ex `fmt.Sprintf("%09d", 12)` ex https://play.golang.org/p/ZREXE0pSq9s

Reply:
Did I understand correctly? 
If notification goes with id, I check that notification contains uid in db: 
1. In case notification doesn't contain uid I should generate it from id and add to db. 
2. If it contains uid, I should use this value further.